pkg/config/config.go

```diff
@@ -100,9 +101,15 @@ func (r IPRouteConfig) Ensure(enabled bool) error {
 func (r IPRuleConfig) Ensure(enabled bool) error {
 	var err error
 	if enabled {
-		err = netlink.RuleAdd(&r.Rule)
-		if os.IsExist(err) {
-			err = nil
+		exist, e := r.exist()
```

Comment:
As we tested on CoS, we would run into a case to add the same rule multiple times. Can we change the logic to ensure the number of desired IP rule to be 1 when enabled is true while 0 when enabled is false?

I think the above logic is more robust as you have no assumption on what the set of IP rules looks like.

Reply:
yeah, I was debating on the same issue. will update.

Reply:
updated.